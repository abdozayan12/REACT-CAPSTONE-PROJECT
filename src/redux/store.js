import { configureStore } from '@reduxjs/toolkit';
import countriesReducer from './countries/countriesSlice';

const store = configureStore({
  reducer: {
    countries: countriesReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: false,
  }),
});

export default store;
