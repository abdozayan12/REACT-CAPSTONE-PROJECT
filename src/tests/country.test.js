import { render, screen } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import CountryBox from '../components/country';
import store from '../redux/store';
import '@testing-library/jest-dom';

describe('CountryBox component test', () => {
  const yemen = {
    flag: 'https://flagcdn.com/w320/ye.png',
    name: 'Yemen',
    pop: 29830000,
    id: 'b235b6cd-63de-461f-97ab-a8cfb49ea39b',
    alt: 'flag of Yemen',
  };

  it('Check if the component changed', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <CountryBox
              key={yemen.id}
              id={yemen.id}
              name={yemen.name}
              flag={yemen.flag}
              pop={yemen.pop}
              alt={yemen.alt}
            />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Check if the component CountryBox container is there', async () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <CountryBox
            key={yemen.id}
            id={yemen.id}
            name={yemen.name}
            flag={yemen.flag}
            pop={yemen.pop}
            alt={yemen.alt}
          />
        </BrowserRouter>
      </Provider>,
    );
    const container = await screen.findByTestId('box');
    expect(container).toBeInTheDocument();
  });
});
