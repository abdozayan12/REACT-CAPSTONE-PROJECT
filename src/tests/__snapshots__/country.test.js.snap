// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryBox component test Check if the component changed 1`] = `
<li
  className="country-box col-span-1 text-white p-4 relative flex flex-col"
  data-testid="box"
  id="b235b6cd-63de-461f-97ab-a8cfb49ea39b"
>
  <button
    onClick={[Function]}
    type="button"
  >
    <svg
      className=" absolute right-2 top-1 text-xl"
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="10"
      />
      <polyline
        points="12 16 16 12 12 8"
      />
      <line
        x1="8"
        x2="16"
        y1="12"
        y2="12"
      />
    </svg>
  </button>
  <div
    className=" h-20 w-28 self-center"
  >
    <img
      alt="flag of Yemen"
      src="https://flagcdn.com/w320/ye.png"
    />
  </div>
  <div
    className="smallDetails self-end"
  >
    <p
      className="countryName font-bold uppercase text-right text-lg sm:text-xl"
    >
      Yemen
    </p>
    <p
      className="population  text-sm"
    >
      population:
       
      <span
        className=" font-bold text-base"
      >
        29.83M
      </span>
    </p>
  </div>
</li>
`;
